File: 题库标准算法/1.两数之和/Python/solution1.1.0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        result = []
        number = {}

        for i in range(len(nums)):
            if nums[i] not in number:
                number[nums[i]] = i

            if target - nums[i] in number and number[target - nums[i]] != i:
                result.append(number[target - nums[i]])
                result.append(i)

                return result

        return result

File: 题库标准算法/1.两数之和/C++/Solution1.1.0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> result(2);
        map<int, unsigned short> number;
        const unsigned short length = nums.size();

        for(unsigned short i = 0; i < length; i++)
        {
            if(number.find(nums[i]) == number.end())
                number[nums[i]] = i;

            if(number.find(target - nums[i]) != number.end() && number[target - nums[i]] != i)
            {
                result[0] = number[target - nums[i]];
                result[1] = i;

                return result;
            }
        }

        return result;
    }
};
